lua/nvim-treesitter-refactor/highlight_definitions.lua

```diff
@@ -9,13 +10,19 @@ local cmd = api.nvim_command
 local M = {}
 
 local usage_namespace = api.nvim_create_namespace('nvim-treesitter-usages')
+local last_node = nil
 
 function M.highlight_usages(bufnr)
-  M.clear_usage_highlights(bufnr)
-
   local node_at_point = ts_utils.get_node_at_cursor()
-  local references = locals.get_references(bufnr)
+  -- Don't calculate usages again if we are on the same node.
+  if node_at_point == last_node and M.has_highlights(bufnr) then
+    return
+  else
+    last_node = node_at_point
```

Comment:
just a doubt: `M.highlight_usages` has a `bufnr` argument, that's say this function is buffer agnostic; so should `last_node` be buffer specified?

Reply:
assume last_node is window level variable, when I switch between buffers, and trigger CursorHold event, so vim will re-call highlight_usages without last_node short circuit, since last_node has changed during switch buffers.

Reply:
oh yeah, this should cache the value per buffer